### PR TITLE
Add compatible targets notifications

### PR DESCRIPTION
**Pull request by PerezIgnacio**
This pull request:
- Adds compatible targets
- Adds related model, request and feature tests
- Adds notifications modal in home page for matches
- Adds channels tests

Trello card: https://trello.com/c/4DBNVg8H/9-5-as-a-user-i-should-receive-notifications-when-a-compatible-target-has-been-created-so-we-can-start-getting-connected

Screen Shot 2019-05-24 at 11 34 22 AM

### CODE REVIEW

File: app/models/target.rb [hunk 1]
```diff
@@ -5,4 +5,30 @@ class Target < ApplicationRecord
   validates :title, :area_length, :latitude, :longitude, presence: true
   validates :latitude, numericality: { greater_than_or_equal_to: -90, less_than_or_equal_to: 90 }
   validates :longitude, numericality: { greater_than_or_equal_to: -180, less_than_or_equal_to: 180 }
+
+  after_save :notify_compatible_targets_users
```
Comment (mdornel) [inline app/models/target.rb]: I wonder if an `after_save` is the best hook possible. Take into account that `notify_compatible_targets_users` will be triggered in any saving action, including any record update.

Maybe `after_create` fits better

File: spec/models/target_spec.rb [hunk 2]
```diff
@@ -1,20 +1,64 @@
...
-  it { should validate_presence_of(:area_length) }
-  it { should validate_presence_of(:latitude) }
-  it { should validate_presence_of(:longitude) }
-  it do
-    should validate_numericality_of(:latitude)
-      .is_greater_than_or_equal_to(-90)
-      .is_less_than_or_equal_to(90)
+describe Target, type: :model do
+  describe 'validations' do
+    it { should validate_presence_of(:title) }
```
Comment (GuilleLeopold) [inline spec/models/target_spec.rb]: use `is_expected` instead of should as this one is deprecated

File: app/views/targets/_match_modal.html.haml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.modal{ id: 'match-modal', tabindex: '-1', role: 'dialog', aria: { hidden: true } }
```
Comment (jpascual1994) [inline app/views/targets/_match_modal.html.haml]: Maybe you can change it to `.modal#match-modal` just to be consistent

File: app/views/targets/_match_modal.html.haml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  .modal-dialog
+    .modal-content
+      = image_tag 'smilies.png', srcset: { 'smilies@2x.png' => '2x', 'smilies@3x.png' => '3x'}, class: 'img-smilies'
+      = t('views.home.match_modal.title')
+      %br/
+      = t('views.home.match_modal.new_match')
+      .match-content
+      %button#start-chat= t('views.home.match_modal.start_chat')
+      = link_to "Skip", 'match-modal', "data-dismiss" => "modal"
```
Comment (jpascual1994) [inline app/views/targets/_match_modal.html.haml]: Use single quotes and also try to use the `data` hash in order to set data attributes
`= link_to 'Skip', 'match-modal', data: { dismiss: 'modal' }`

**Pull request merged**